polyfill/lib/timezone.mjs

```diff
@@ -73,20 +87,30 @@ export class TimeZone {
       }
     }
 
-    const utcns = ES.GetEpochFromParts(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond);
+    const utcns = ES.GetEpochFromParts(
+      GetSlot(dateTime, YEAR),
+      GetSlot(dateTime, MONTH),
+      GetSlot(dateTime, DAY),
+      GetSlot(dateTime, HOUR),
+      GetSlot(dateTime, MINUTE),
+      GetSlot(dateTime, SECOND),
+      GetSlot(dateTime, MILLISECOND),
+      GetSlot(dateTime, MICROSECOND),
+      GetSlot(dateTime, NANOSECOND)
+    );
     if (utcns === null) throw new RangeError('DateTime outside of supported range');
     const before = ES.GetTimeZoneOffsetNanoseconds(utcns.minus(bigInt(86400 * 1e9)), GetSlot(this, IDENTIFIER));
     const after = ES.GetTimeZoneOffsetNanoseconds(utcns.plus(bigInt(86400 * 1e9)), GetSlot(this, IDENTIFIER));
-    const nanoseconds = after.minus(before);
+    const nanoseconds = after - before;
     const diff = ES.ToTemporalDurationRecord({ nanoseconds }, 'reject');
     switch (disambiguation) {
       case 'earlier': {
         const earlier = dateTime.minus(diff);
-        return this.getAbsoluteFor(earlier, disambiguation);
+        return this.getPossibleAbsolutesFor(earlier)[0];
       }
       case 'later': {
         const later = dateTime.plus(diff);
-        return this.getAbsoluteFor(later, disambiguation);
+        return this.getPossibleAbsolutesFor(later)[0];
```

Comment:
Does this really match the previous behavior?

Reply:
Hmm, I thought it didn't matter, but I see now it could make a difference if a time zone had two transitions immediately adjacent to each other.

Reply:
Fixed this case, I think.